service/worker/scanner/history/scavenger.go

```diff
@@ -96,18 +100,20 @@ func NewScavenger(
 	hbd ScavengerHeartbeatDetails,
 	metricsClient metrics.Client,
 	logger log.Logger,
+	maxWorkflowRetentionInDays int,
```

Comment:
nit: shall we make this really dynamic and doesn't require worker restart upon changing the value?

Reply:
good point.

Reply:
Done. Thanks for the comments :D